src/app/WriteClient.cpp

```diff
@@ -350,7 +359,9 @@ CHIP_ERROR WriteClientHandle::SendWriteRequest(SessionHandle session, System::Cl
 {
     CHIP_ERROR err = mpWriteClient->SendWriteRequest(session, timeout);
 
-    if (err == CHIP_NO_ERROR)
+    // The InteractionModelEngine will only close the writeClienHandle upon reception of a writeResponse
+    // This won't happen in a Groupcast communication.
+    if (err == CHIP_NO_ERROR && !session.IsGroupSession())
```

Comment:
Do we really need the `SetWriteClient(nullptr)` check below? Wouldn't that happen naturally as part of the destruction of the handle?

@erjiaqing

Reply:
Since this code was already there (and working) I would leave it as is since it was done prior to this PR.